internal/reconstitution/controller.go

```diff
@@ -105,3 +117,45 @@ func (r *controller) populateCache(ctx context.Context, comp *apiv1.Composition,
 
 	return r.Cache.fill(ctx, comp, synthesis, slices)
 }
+
+func (r *controller) HandleReadinessTransition(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	slice := &apiv1.ResourceSlice{}
+	err := r.client.Get(ctx, req.NamespacedName, slice)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("unable to get resource slice: %w", err))
+	}
+
+	owner := metav1.GetControllerOf(slice)
+	if owner == nil {
+		return ctrl.Result{}, nil
+	}
+
+	for i, res := range slice.Status.Resources {
+		if res.Ready == nil {
+			continue // only care about resources that have become ready
+		}
+
+		res, ok := r.Cache.getByIndex(&sliceIndex{
+			Index:     i,
+			SliceName: slice.Name,
+			Namespace: slice.Namespace,
+		})
+		if !ok {
+			logger.V(1).Info("a dependent resource was not found in cache - this is unexpected")
+			return ctrl.Result{}, nil
+		}
+
+		synRef := &SynthesisRef{CompositionName: owner.Name, Namespace: req.Namespace, UUID: slice.Spec.SynthesisUUID}
+		resources := r.Cache.RangeByReadinessGroup(ctx, synRef, res.ReadinessGroup, 1)
```

Comment:
Const (and maybe type?) for the direction? It's a bit hard to read at first glance